packages/authorization-process/test/addClientPurpose.test.ts

```diff
@@ -0,0 +1,557 @@
+      interface: getMockDocument(),
+      publishedAt: new Date(),
+    };
+
+    const mockEservice = {
+      ...getMockEService(),
+      descriptors: [mockDescriptor],
+    };
+    const mockConsumer = getMockTenant();
+
+    const mockPurpose: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEservice.id,
+      consumerId: generateId(),
+      versions: [getMockPurposeVersion(purposeVersionState.active)],
+    };
+
+    const mockClient: Client = {
+      ...getMockClient(),
+      consumerId: mockConsumer.id,
+    };
+
+    const mockAgreement: Agreement = {
+      ...getMockAgreement(),
+      state: agreementState.active,
+      eserviceId: mockEservice.id,
+      descriptorId: mockDescriptor.id,
+      consumerId: mockConsumer.id,
+    };
+
+    await addOneClient(mockClient);
+    await writeInReadmodel(toReadModelPurpose(mockPurpose), purposes);
+    await writeInReadmodel(toReadModelEService(mockEservice), eservices);
+    await writeInReadmodel(toReadModelAgreement(mockAgreement), agreements);
+
+    expect(
+      authorizationService.addClientPurpose({
+        clientId: mockClient.id,
+        seed: { purposeId: mockPurpose.id },
+        organizationId: mockConsumer.id,
+        correlationId: generateId(),
+        logger: genericLogger,
+      })
+    ).rejects.toThrowError(
+      organizationNotAllowedOnPurpose(mockConsumer.id, mockPurpose.id)
+    );
+  });
+  it("should throw eserviceNotFound if the eservice doesn't exist", async () => {
+    const mockDescriptor: Descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.published,
+      interface: getMockDocument(),
+      publishedAt: new Date(),
+    };
+
+    const mockEservice = {
+      ...getMockEService(),
+      descriptors: [mockDescriptor],
+    };
+    const mockConsumer = getMockTenant();
+
+    const mockPurpose: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEservice.id,
+      consumerId: mockConsumer.id,
+      versions: [getMockPurposeVersion(purposeVersionState.active)],
+    };
+
+    const mockClient: Client = {
+      ...getMockClient(),
+      consumerId: mockConsumer.id,
+    };
+
+    const mockAgreement: Agreement = {
+      ...getMockAgreement(),
+      state: agreementState.active,
+      eserviceId: mockEservice.id,
+      descriptorId: mockDescriptor.id,
+      consumerId: mockConsumer.id,
+    };
+
+    await addOneClient(mockClient);
+    await writeInReadmodel(toReadModelPurpose(mockPurpose), purposes);
+    await writeInReadmodel(toReadModelAgreement(mockAgreement), agreements);
+
+    expect(
+      authorizationService.addClientPurpose({
+        clientId: mockClient.id,
+        seed: { purposeId: mockPurpose.id },
+        organizationId: mockConsumer.id,
+        correlationId: generateId(),
+        logger: genericLogger,
+      })
+    ).rejects.toThrowError(eserviceNotFound(mockEservice.id));
+  });
+  it("should throw agreementNotFound if the agreement doesn't exist", async () => {
```

Comment:
```suggestion
  it("should throw agreementNotFound if there is no agreement in required states", async () => {
```
we could add the `it.each` to test all cases

Reply:
Updated. Now all the non-compatible states are covered. Also, the case of no agreement at all is covered as well.